test/e2e/metrics_k8s_cluster_receiver_test.go

```diff
@@ -0,0 +1,154 @@
+			objs = append(objs, backendContainerMetricsEnabled.K8sObjects()...)
+			backendK8sClusterContainerMetricsEnabledURL = backendContainerMetricsEnabled.ExportURL(proxyClient)
+
+			pipelineOnlyContainerMetricsEnabled := testutils.NewMetricPipelineBuilder().
+				WithName(pipelineK8sClusterContainerMetricsEnabledName).
+				WithRuntimeInput(true).
+				WithRuntimeInputPodMetrics(false).
+				WithOTLPOutput(testutils.OTLPEndpoint(backendContainerMetricsEnabled.Endpoint())).
+				Build()
+			objs = append(objs, &pipelineOnlyContainerMetricsEnabled)
+
+			backendPodMetricsEnabled := backend.New(mockNs, backend.SignalTypeMetrics, backend.WithName(backendK8sClusterPodMetricsEnabledName))
+			objs = append(objs, backendPodMetricsEnabled.K8sObjects()...)
+			backendK8sClusterPodMetricsEnabledURL = backendPodMetricsEnabled.ExportURL(proxyClient)
+
+			pipelineOnlyPodMetricsEnabled := testutils.NewMetricPipelineBuilder().
+				WithName(pipelineK8sClusterPodMetricsEnabledName).
+				WithRuntimeInput(true).
+				WithRuntimeInputContainerMetrics(false).
+				WithOTLPOutput(testutils.OTLPEndpoint(backendPodMetricsEnabled.Endpoint())).
+				Build()
+			objs = append(objs, &pipelineOnlyPodMetricsEnabled)
+
+			return objs
+		}
+
+		BeforeAll(func() {
+			k8sObjects := makeResources()
+
+			DeferCleanup(func() {
+				Expect(kitk8s.DeleteObjects(ctx, k8sClient, k8sObjects...)).Should(Succeed())
+			})
+			Expect(kitk8s.CreateObjects(ctx, k8sClient, k8sObjects...)).Should(Succeed())
+		})
+
+		It("Should have healthy pipelines", func() {
+			assert.MetricPipelineHealthy(ctx, k8sClient, pipelineK8sClusterContainerMetricsEnabledName)
+			assert.MetricPipelineHealthy(ctx, k8sClient, pipelineK8sClusterPodMetricsEnabledName)
+		})
+
+		It("Should have metrics backends running", func() {
+			assert.DeploymentReady(ctx, k8sClient, types.NamespacedName{Name: backendK8sClusterContainerMetricsEnabledName, Namespace: mockNs})
+			assert.DeploymentReady(ctx, k8sClient, types.NamespacedName{Name: backendK8sClusterPodMetricsEnabledName, Namespace: mockNs})
+		})
+
+		It("Should deliver runtime container metrics to container-metrics backend", func() {
+			Eventually(func(g Gomega) {
+				resp, err := proxyClient.Get(backendK8sClusterContainerMetricsEnabledURL)
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(resp).To(HaveHTTPStatus(http.StatusOK))
+				bodyContent, err := io.ReadAll(resp.Body)
+				defer resp.Body.Close()
+				g.Expect(err).NotTo(HaveOccurred())
+
+				containerMetricNames := slices.Concat(kubeletstats.ContainerMetricsNames, k8scluster.ContainerMetricsNames)
+				g.Expect(bodyContent).To(WithFlatMetrics(WithNames(
+					OnlyContainElementsOf(containerMetricNames))), "Found container metrics in backend that are not part of k8scluster or kubeletstats")
+
+				g.Expect(bodyContent).To(WithFlatMetrics(
+					ContainElement(HaveField("Name", BeElementOf(k8scluster.ContainerMetricsNames)))), "Found container metrics in backend that are not part of k8scluster")
```

Comment:
We are checking here if we find any metrics that is part of the k8scluster container metrics. So maybe the error message can be something like this: `Did not find any k8scluster container metrics in backend`